src/rollback_to_stable/rts_api.c

```diff
@@ -142,15 +142,23 @@ __rollback_to_stable_int(WT_SESSION_IMPL *session, bool no_ckpt)
      * though the stable timestamp isn't supposed to be updated while rolling back, accessing it
      * without a lock would violate protocol.
      */
-    WT_ACQUIRE_READ_WITH_BARRIER(rollback_timestamp, txn_global->stable_timestamp);
+    WT_ACQUIRE_READ_WITH_BARRIER(stable_timestamp, txn_global->stable_timestamp);
     WT_ACQUIRE_READ_WITH_BARRIER(pinned_timestamp, txn_global->pinned_timestamp);
     __wt_verbose_multi(session, WT_VERB_RECOVERY_RTS(session),
       WT_RTS_VERB_TAG_INIT
       "start rollback to stable with stable_timestamp=%s and oldest_timestamp=%s using %u worker "
       "threads",
-      __wt_timestamp_to_string(rollback_timestamp, ts_string[0]),
+      __wt_timestamp_to_string(stable_timestamp, ts_string[0]),
       __wt_timestamp_to_string(txn_global->oldest_timestamp, ts_string[1]), threads);
 
+    /* If the stable timestamp is not set, do not roll back based on it. */
+    rollback_timestamp = stable_timestamp;
+    if (rollback_timestamp == WT_TS_NONE) {
```

Comment:
Why not do:

```
      if (stable_timestamp == WT_TS_NONE) {
            rollback_timestamp = WT_TS_MAX;
            __wt_verbose_multi(session, WT_VERB_RECOVERY_RTS(session), WT_RTS_VERB_TAG_NO_STABLE "%s",
          "the stable timestamp is not set; set the rollback timestamp to the maximum timestamp");
      } else
            rollback_timestamp = stable_timestamp;
```

Reply:
I changed this in [edf1709](https://github.com/wiredtiger/wiredtiger/pull/10580/commits/edf1709996c9292dc8d2ec2cba263432c0feb89c); either way works for me.